include/wd_aead.h

```diff
@@ -0,0 +1,147 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+#ifndef __WD_AEAD_H
+#define __WD_AEAD_H
+
+#include <dlfcn.h>
+#include "wd_alg_common.h"
+#include "config.h"
+#include "wd_cipher.h"
+#include "wd_digest.h"
+#include "wd.h"
+
+/**
+ * wd_aead_op_type - Algorithm type of option
+ */
+enum wd_aead_op_type {
+	WD_CIPHER_ENCRYPTION_DIGEST,
+	WD_CIPHER_DECRYPTION_DIGEST,
+	WD_DIGEST_CIPHER_ENCRYPTION,
+	WD_DIGEST_CIPHER_DECRYPTION,
+};
+
+/**
+ * wd_cipher_type - Algorithm type of cipher
+ * statement in wd_cipher.h
+ *
+ * wd_cipher_mode - Algorithm mode of cipher
+ * statement in wd_cipher.h
+ */
+
+struct wd_aead_sess_setup {
+	enum wd_cipher_alg calg;
+	enum wd_cipher_mode cmode;
+	enum wd_digest_type dalg;
+	enum wd_digest_mode dmode;
+};
+
+struct wd_aead_req;
+typedef void *wd_alg_aead_cb_t(struct wd_aead_req *req, void *cb_param);
+
+struct wd_aead_sess {
+	char			*alg_name;
+	enum wd_cipher_alg	calg;
+	enum wd_cipher_mode	cmode;
+	enum wd_digest_type	dalg;
+	enum wd_digest_mode	dmode;
+	void			*ckey;
+	void			*akey;
+	__u16			ckey_bytes;
+	__u16			akey_bytes;
+	__u16			auth_bytes;
+	void			*priv;
+};
+
+struct wd_aead_req {
+	enum wd_aead_op_type op_type;
+	void			*src;
+	void			*dst;
+	void			*iv;
+
+	__u32			in_bytes;
+	__u32			out_bytes;
+	__u32			out_buf_bytes;
+	__u16			iv_bytes;
+	__u16			assoc_bytes;
+	__u16			state;
+	wd_alg_aead_cb_t	*cb;
+	void			*cb_param;
+};
+
+/**
+ * wd_aead_init() Initialise ctx configuration and schedule.
+ * @ config	    User defined ctx configuration.
+ * @ sched	    User defined schedule.
+ */
+int wd_aead_init(struct wd_ctx_config *config, struct wd_sched *sched);
+void wd_aead_uninit(void);
+
+/**
+ * wd_aead_alloc_sess() Allocate a wd aead session
+ * @ setup Parameters to setup this session.
+ */
+handle_t wd_aead_alloc_sess(struct wd_aead_sess_setup *setup);
+
+/**
+ * wd_aead_free_sess()
+ * @ sess, need to be freed sess
+ */
+void wd_aead_free_sess(handle_t h_sess);
+
+/**
+ * wd_aead_set_ckey() Set cipher key to aead session.
+ * @h_sess: wd aead session.
+ * @key: cipher key addr.
+ * @key_len: cipher key length.
```

Comment:
add return value decription, and also for other return value

Reply:
The return value description is in the interface document, no need to repeat the description here